client/pages/home.jsx

```diff
@@ -1,31 +1,68 @@
 import React from 'react';
 import PostList from './post-list';
-
+import Modal from '../components/modal';
 export default class Home extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      clicked: false,
-      posts: []
+      postId: false,
+      posts: [],
+      open: false,
+      addedComment: false
     };
-    this.openEvent = this.openEvent.bind(this);
+    this.addComment = this.addComment.bind(this);
+    this.resetState = this.resetState.bind(this);
+  }
+
+  addComment(postId) {
+    this.setState({
+      postId: postId,
+      open: true
+    });
   }
 
-  openEvent() {
-    this.setState(prevState => ({
-      clicked: !prevState.clicked
-    }));
+  resetState() {
+    this.setState({
+      open: false,
+      addedComment: true
+    });
   }
 
   componentDidMount() {
-    fetch('/api/posts/')
-      .then(response => response.json())
-      .then(posts => this.setState({ posts: posts }));
+    Promise.all([
+      fetch('/api/posts/'),
+      fetch('/api/comments/')
+    ])
+      .then(([res1, res2]) => Promise.all([res1.json(), res2.json()]))
+      .then(([data1, data2]) => this.setState({
+        posts: data1,
+        comments: data2
+      }));
+  }
+
+  componentDidUpdate() {
+    if (this.state.addedComment) {
+      Promise.all([
+        fetch('/api/posts/'),
+        fetch('/api/comments/')
+      ])
+        .then(([res1, res2]) => Promise.all([res1.json(), res2.json()]))
+        .then(([data1, data2]) => this.setState({
+          posts: data1,
+          comments: data2
+        }));
+      this.setState({
+        addedComment: false
+      });
+    }
```

Comment:
You really don't want to refetch the entire list of posts and comments. Try to program your server to give you the added comment back including the username, then append it to your comments list like in fetch-in-react-practice, this is way too much network activity for a simple update.